TypeCobol/Compiler/Directives/TypeCobolOptions.cs

```diff
@@ -99,5 +108,54 @@ public TypeCobolOptions Clone()
             this.CopyIBMOptionsTo(newOptions);
             return newOptions;
         }
+
+#if EUROINFO_RULES
+        internal void CollectUsedCopy(CopyDirective copy)
+        {
+            if (ReportUsedCopyNamesPath != null)
+            {
+                if (CollectedCopyNames == null)
+                    CollectedCopyNames = new System.Collections.Generic.Dictionary<string, System.Collections.Generic.HashSet<string>>();
+
+                string copyName = copy.TextName.ToUpper();
+                if (!CollectedCopyNames.ContainsKey(copyName))
+                {
+                    CollectedCopyNames[copyName] = new System.Collections.Generic.HashSet<string>();
+                }
+                if (copy.Suffix != null)
+                {
+                    System.Collections.Generic.HashSet<string> suffixedNames = CollectedCopyNames[copyName];
+                    suffixedNames.Add(copyName + copy.Suffix.ToUpper());
+                }                
+            }
+        }
+
+        /// <summary>
+        /// Report all Used Copy Name with suffixed names if any
+        /// </summary>
+        public void ReportCollectedUsedCopy()
+        {
+            if (ReportUsedCopyNamesPath != null)
+            {
+                using (System.IO.StreamWriter sw = new System.IO.StreamWriter(ReportUsedCopyNamesPath))
+                {
+                    if (CollectedCopyNames != null)
+                    {
+                        foreach (string cpyName in CollectedCopyNames.Keys)
+                        {
+                            sw.Write(cpyName);
+                            System.Collections.Generic.HashSet<string> suffixedNames = CollectedCopyNames[cpyName];
+                            foreach (string suffixed in suffixedNames)
+                            {
+                                sw.Write(';');
+                                sw.Write(suffixed);
+                            }
+                            sw.WriteLine();
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
This method should be moved into CLI as it is more focused on file manipulation. Also it would avoid the duplicate nullity check on the file path.

Reply:
Yes it's true but the purpose here is to be able to collect information out of CLI class, using only TypeCobol.dll